src/generic/GenericTypes.jl

```diff
@@ -1099,11 +1099,16 @@ struct MatSpaceView{T <: NCRingElement, V, W} <: Mat{T}
    base_ring::NCRing
 end
 
-struct MatSpaceVecView{T <: NCRingElement, V, W} <: AbstractVector{T}
+struct MatSpaceVecView{T <: NCRingElement, V, W} <: AbstractArray{T, 1}
```

Comment:
```suggestion
struct MatSpaceVecView{T <: NCRingElement, V, W} <: AbstractVector{T}
```
I'd prefer to keep this an `AbstractVector` as we use that everywhere else whenever possible (unless we overlooked something)

Reply:
Sounds good.  I just thought it was nice to see the difference between `MatSpaceVecView` and `MatSpacePointView`, but given that it is written elsewhere, I agree.